src/ui/Facet/FacetSearchElement.ts

```diff
@@ -0,0 +1,139 @@
+
+    this.wait = document.createElement('div');
+    this.wait.innerHTML = SVGIcons.icons.loading;
+    $$(this.wait).addClass('coveo-facet-search-wait-animation');
+    SVGDom.addClassToSVGInContainer(this.wait, 'coveo-facet-search-wait-animation-svg');
+    this.search.appendChild(this.wait);
+    this.hideFacetSearchWaitingAnimation();
+
+    this.clear = $$(
+      'div',
+      { className: 'coveo-facet-search-clear', title: l('Clear', l('Search')) },
+      SVGIcons.icons.checkboxHookExclusionMore
+    ).el;
+    SVGDom.addClassToSVGInContainer(this.clear, 'coveo-facet-search-clear-svg');
+    this.clear.style.display = 'none';
+    this.search.appendChild(this.clear);
+
+    const middle = document.createElement('div');
+    $$(middle).addClass('coveo-facet-search-middle');
+    this.search.appendChild(middle);
+
+    this.input = document.createElement('input');
+    this.input.setAttribute('type', 'text');
+    this.input.setAttribute('autocapitalize', 'off');
+    this.input.setAttribute('autocorrect', 'off');
+    $$(this.input).addClass('coveo-facet-search-input');
+    Component.pointElementsToDummyForm(this.input);
+    middle.appendChild(this.input);
+
+    $$(this.input).on('keyup', (e: KeyboardEvent) => {
+      handleFacetSearchKeyUp(e);
+    });
+    $$(this.clear).on('click', (e: Event) => {
+      handleFacetSearchClear();
+    });
+    $$(this.input).on('focus', (e: Event) => {
+      handleFacetSearchFocus();
+    });
+
+    this.detectSearchBarAnimation();
+    return this.search;
+  }
+
+  public showFacetSearchWaitingAnimation() {
+    $$(this.magnifier).hide();
+    $$(this.wait).show();
+  }
+
+  public hideFacetSearchWaitingAnimation() {
+    $$(this.magnifier).show();
+    $$(this.wait).hide();
+  }
+
+  public detectSearchBarAnimation() {
+    EventsUtils.addPrefixedEvent(this.search, 'AnimationStart', event => {
+      if (event.animationName == 'grow') {
+        this.searchBarIsAnimating = true;
+      }
+    });
+
+    EventsUtils.addPrefixedEvent(this.search, 'AnimationEnd', event => {
+      if (event.animationName == 'grow') {
+        this.searchBarIsAnimating = false;
+      }
+    });
+  }
+
+  public positionSearchResults(root: HTMLElement, facetWidth: number, nextTo: HTMLElement) {
+    if (this.searchResults != null) {
+      root.appendChild(this.searchResults);
+      this.searchResults.style.display = 'block';
+      this.searchResults.style.width = facetWidth - 40 + 'px';
+
+      if ($$(this.searchResults).css('display') == 'none') {
+        this.searchResults.style.display = '';
+      }
+      let searchBar = $$(this.search);
+      if (searchBar.css('display') == 'none' || this.searchBarIsAnimating) {
+        if ($$(this.searchResults).css('display') == 'none') {
+          this.searchResults.style.display = '';
+        }
+        EventsUtils.addPrefixedEvent(this.search, 'AnimationEnd', evt => {
+          PopupUtils.positionPopup(this.searchResults, nextTo, root, {
```

Comment:
Removing the duplication of this method here and on line 122. Maybe an anonymous function stored in a varibale in the same method would do, or breaking out into a separate method?

Also the `evt` variable on line 114 does not seem to be used. If so, I think it's better to use `()`.